libtool-testsuite: Add test-dependencies and skip FC-tests if not present

bernhardkaindl opened the pull request:
@adamjstewart - maybe you could approve this?
    
- The testsuite of 2.4.6 and newer needs autoconf >= 2.62
- The tests use automake, add it if building with --test
- If Fortran compilers are not present, skip the tests which use them.

bernhardkaindl commented:
@spackbot run pipeline

bernhardkaindl commented:
@nhanford I just let the pipeline run and got at unexpected failure message:
```pt
$ spack --version
0.16.2
...
Error: Expected database version 5 but found version 6.
`spack reindex` may fix this, or you may need a newer Spack version.`
```
Could this be due to your patch being merged now?
```py
-_db_version = Version('5')
+_db_version = Version('6')
```
> @scottwittenburg Since you knew where to update the pinned OpenSSL version from 1.1.1k to 1.1.1l, maybe you know about how this database mismatch could be fixed, it appers to come from commit d83f7110d56 from @nhanford today (see database version update from 5 to 6 in his patch above)

Doh, my branch wasn't rebased to develop, so this was likely the cause. Rebased to be run again.
```py
$ spack --version
0.16.2
$ cd share/spack/gitlab/cloud_pipelines/stacks/${SPACK_CI_STACK_NAME}
$ spack env activate --without-view .
$ spack ci generate --check-index-only --artifacts-root "${CI_PROJECT_DIR}/jobs_scratch_dir" --output-file "${CI_PROJECT_DIR}/jobs_scratch_dir/cloud-ci-pipeline.yml"
==> Error: Expected database version 5 but found version 6.
`spack reindex` may fix this, or you may need a newer Spack version.
```

tgamblin commented:
@bernhardkaindl can you see if rebasing on develop fixes this for you? I *think* this is because the buildcache indices updated with the merge of @nhanford’s PR, after this pipeline was generated with a prior merge commit. I can’t verify this right now as I’m on a phone and just woke up.

Not sure how best to fix this type of issue in CI, as the buildcache DB versions aren’t directly synced with branches right now. We could potentially do a follow-up PR to retain multiple, versioned buildcache indices for transitions like this.

bernhardkaindl commented:
> @bernhardkaindl can you see if rebasing on develop fixes this for you?

Doh! Rebased it to develop and pushed.

Currently the ci/gitalb-ci Pipeline is backlogged now in this and  #25790, I'll see later.

bernhardkaindl committed:
libtool: fix running the unit-tests with spack install --test root

Besides adding autoconf and automake as needed for tests of 2.4.6,
skip Fortran test cases when Fortran compilers are not provided.

alalazo commented:
@spackbot run pipeline
